torchtext/csrc/vocab.cpp

```diff
@@ -125,18 +157,24 @@ static auto vocab =
         .def("get_stoi", &Vocab::get_stoi)
         .def("get_itos", &Vocab::get_itos)
         .def_pickle(
-            // __getstate__
-            [](const c10::intrusive_ptr<Vocab> &self)
-                -> std::tuple<std::vector<std::string>, std::string> {
-              std::tuple<std::vector<std::string>, std::string> states(
-                  self->itos_, self->unk_token_);
+            // __setstate__
+            [](const c10::intrusive_ptr<Vocab> &self) -> VocabStates {
+              std::string version_str = "0.0.1";
```

Comment:
You could turn this into a function "_get_states_from_vocab" or such. Then we have the save and load both abstracted into separate functions.